packages/block-editor/src/components/rich-text/index.js

```diff
@@ -127,6 +127,19 @@ export class RichText extends Component {
 			this.onSplit = this.props.unstableOnSplit;
 		}
 
+		if ( this.props.disallowFormats ) {
+			this.disallowFormats = this.props.disallowFormats;
+		} else if ( this.props.formattingControls ) {
+			deprecated( 'wp.editor.RichText formattingControls prop', {
+				plugin: 'Gutenberg',
+				alternative: 'wp.editor.RichText disallowFormats prop',
+			} );
+
+			if ( this.props.formattingControls.length === 0 ) {
+				this.disallowFormats = true;
+			}
```

Comment:
This isn't completely backwards compatible. Would something like this work?

```js
this.disallowFormats = difference( [ 'bold', 'italic', 'link', 'strikethrough' ], this.props.formattingControls );
```

Reply:
I changed a few things now so there's only a withoutInteractiveFormatting prop. This will no longer work.